pkg/gatherers/clusterconfig/openshift_machine_api_events.go

```diff
@@ -0,0 +1,78 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// GatherOpenshiftMachineApiEvents collects warning ("abnormal") events
+// from "openshift-machine-api" namespace
+//
+// *Location of events in archive: config/events/
+func (g *Gatherer) GatherOpenshiftMachineApiEvents(ctx context.Context) ([]record.Record, []error) {
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+	namespace := "openshift-machine-api"
+	records, err := gatherOpenshiftMachineApiEvents(ctx, gatherKubeClient.CoreV1(), namespace, g.interval)
+	if err != nil {
+		return nil, []error{err}
+	}
+	return records, nil
+}
+
+func gatherOpenshiftMachineApiEvents(ctx context.Context,
+	coreClient corev1client.CoreV1Interface,
+	namespace string,
```

Comment:
You called the function `gatherOpenshiftMachineApiEvents` so no need to pass the `namespace` name as parameter. I think you can simply hardcoded here `coreClient.Events("openshift-machine-api").List(ctx, metav1.ListOptions{})`

Reply:
Done.